### PR TITLE
Adding queryables link to landing page

### PR DESCRIPTION
**Pull request by constantinius**
**Related Issue(s):**

- #401

**Description:**

Adds the `/queryables` link to the landing page

Requires https://github.com/stac-utils/stac-pydantic/pull/123

**PR Checklist:**

- [ ] `pre-commit` hooks pass locally
- [x] Tests pass (run `make test`)
- [ ] Documentation has been updated to reflect changes, if applicable, and docs build successfully (run `make docs`)
- [x] Changes are added to the [CHANGELOG](https://github.com/stac-utils/stac-fastapi/blob/main/CHANGES.md).

### CODE REVIEW

File: stac_fastapi/types/stac_fastapi/types/core.py [hunk 1]
```diff
@@ -306,6 +306,13 @@ def _landing_page(
                     "href": urljoin(base_url, "search"),
                     "method": "POST",
                 },
+                {
```
Comment (m-mohr) [inline stac_fastapi/types/stac_fastapi/types/core.py:309]: Is this always available? Maybe only expose it if the filter extension is enabled and there are actually any queryables available?

Comment (constantinius) [reply stac_fastapi/types/stac_fastapi/types/core.py:309]: Unfortunately not, the API routes are only added via the filter extension here: https://github.com/stac-utils/stac-fastapi/blob/main/stac_fastapi/extensions/stac_fastapi/extensions/core/filter/filter.py#L85

There is currently no way to register link objects to API objects like the landing page or collections. Not sure how to proceed here.

Comment (m-mohr) [reply stac_fastapi/types/stac_fastapi/types/core.py:309]: That would be useful in general: filter, language, collection-search and potentially also sort and query all may add links.

Comment (gadomski) [reply stac_fastapi/types/stac_fastapi/types/core.py:309]: The relationship between the returned data structures (e.g. the landing page) and links has been a pain point for a while, and was one of the issues that derailed the original backend breakout over a year ago; see https://github.com/stac-utils/stac-fastapi/pull/472 for one of the PRs in that space.

I personally haven't given the issue too much direct thought, so prototype solutions are welcome @constantinius. N.B. I'm no longer being supported to help maintain this project, so unfortunately reviews might come at a slower pace as I'm currently doing them on personal time 🙇🏼.

Comment (gadomski) [reply stac_fastapi/types/stac_fastapi/types/core.py:309]: @constantinius looking at this again (after a while), it looks like you could check for the filter extension in `BaseCoreClient` and `AsyncBaseCoreClient`, e.g. in: https://github.com/stac-utils/stac-fastapi/blob/ded2283f74d4f5722dd251a45c448e842ae9f219/stac_fastapi/types/stac_fastapi/types/core.py#L354

There's a `extensions` attribute on the client.

Comment (vincentsarago) [reply stac_fastapi/types/stac_fastapi/types/core.py:311]: in titiler-pgstac we test to see if the endpoint exist https://github.com/stac-utils/titiler-pgstac/blob/ffe0b68c56eac9c2d5040d61a9022d4705807881/titiler/pgstac/factory.py#L1006-L1020

**Comment by vincentsarago** [comment]
@jonhealy1 This is the only PR which needs to be merged before 2.5 release!

**Review by jonhealy1 — APPROVED** [review]
LGTM

**Comment by m-mohr** [comment]
Didn't this PR require https://github.com/stac-utils/stac-pydantic/pull/123, which is still open?

**Comment by vincentsarago** [comment]
😬 because this PR wasn't marked as Draft I didn't know it was dependent to another PR in stac-pydantic 😓

**Pull request merged**